johnzon-core/src/test/java/org/apache/johnzon/core/JsonParserStreamingTest.java

```diff
@@ -121,4 +127,59 @@ private String parserAndConcat(String json) {
                 .collect(Collectors.joining(","));
         return sum;
     }
+
+    @Test
+    public void testGetArrayStream() {
+        StringReader sr = new StringReader("[1,2,3,4,5,6]");
+        JsonParser jsonParser = Json.createParser(sr);
+
+        JsonParser.Event firstEvent = jsonParser.next();
+        assertEquals(JsonParser.Event.START_ARRAY, firstEvent);
+
+        int sum = jsonParser.getArrayStream()
+                .mapToInt(v -> ((JsonNumber)v).intValue())
+                .sum();
+        assertEquals(21, sum);
+    }
+
+    @Test(expected = JsonParsingException.class)
+    public void testParseErrorInGetArrayStream() {
+        StringReader sr = new StringReader("[\"this is\":\"not an object\"]");
+        JsonParser jsonParser = Json.createParser(sr);
+
+        JsonParser.Event firstEvent = jsonParser.next();
+        assertEquals(JsonParser.Event.START_ARRAY, firstEvent);
+
+        jsonParser.getArrayStream().forEach(dummy -> {});
+    }
+
+    @Test
+    public void testGetObjectStream() {
+        StringReader sr = new StringReader("{\"foo\":\"bar\",\"baz\":\"quux\",\"something\":\"else\"}");
+        JsonParser jsonParser = Json.createParser(sr);
+
+        JsonParser.Event firstEvent = jsonParser.next();
+        assertEquals(JsonParser.Event.START_OBJECT, firstEvent);
+
+        Map<String, String> mappings = jsonParser.getObjectStream()
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> ((JsonString)e.getValue()).getString()));
+
+        Map<String, String> expectedMappings = new HashMap<>();
+        expectedMappings.put("foo", "bar");
+        expectedMappings.put("baz", "quux");
+        expectedMappings.put("something", "else");
+        assertEquals(expectedMappings, mappings);
+    }
+
+    @Test(expected = JsonParsingException.class)
+    public void testParseErrorInGetObjectStream() {
+        StringReader sr = new StringReader("{42}");
+        JsonParser jsonParser = Json.createParser(sr);
+
+        JsonParser.Event firstEvent = jsonParser.next();
```

Comment:
guess we should use a try with resource pattern to close the parser (it is required even for string readers due to memory management, even if not critical in tests it is always better to not show antipatterns in tests since they are kind of part of the doc)
if you test the message it looks ok.

But looks sufficient for me as coverage.

Reply:
None of the other tests closed the parser, that's why I assumed it's fine. I've changed now so the parser is closed not just in the new tests, but the existing ones as well.